### PR TITLE
fix(sequencer): fix fee estimation

### LINKED ISSUE #1711: sequencer fee estimation fails on missing fees
Currently, the sequencer fee estimation in `get_fees_for_transaction()` fails if any action is missing fees, which makes our setup of having actions disabled via missing fees pointless. We should change the fee calculation to only fail if an action within that transaction is missing fees.

https://github.com/astriaorg/astria/blob/5ff422f7cecd690684f47090bd6cd6395830441c/crates/astria-sequencer/src/transaction/checks.rs#L63-L66

┆Issue Number: ENG-943

### PR DESCRIPTION
**Pull request by ethanoroshiba**
## Summary
Changed `get_fees_for_transaction` to reflect recent changes to fee logic action disable/enablement.

## Background
Fee estimation was still failing upon there being missing fees for any action, even though it's okay for there to be no fees, as long as the disabled action is not part of the transaction being calculated.

## Changes
- Only check if fees are present on a per-action basis.
- Use single fee calculation function for all fees.

## Testing
Passing all in-code tests, manual testing to come.

## Related Issues
closes #1711

### CODE REVIEW

**Review by Fraser999 — APPROVED** [review]

**Review by SuperFluffy — APPROVED** [review]
Approving because it's late in the cycle, but noticing a couple of code quality things that might make debugging things a pain.

File: crates/astria-sequencer/src/fees/action.rs [hunk 1]
```diff
@@ -133,135 +133,98 @@ impl ActionHandler for FeeAssetChange {
...
-            base: 2,
-            multiplier: 0,
-        });
-
-        fee_change.check_and_execute(&mut state).await.unwrap();
-        assert_eq!(state.get_ics20_withdrawal_fees().await.unwrap().base, 2);
+    /// This macro generates a test named e.g. `transfer_fee_change_action_executes` which asserts
+    /// that executing a `FeeChange` tx for the given action results in the fees being stored for
+    /// the given action.
+    macro_rules! test_fee_change_action {
```
[the remark this answers is not in the document]
Comment (ethanoroshiba) [reply crates/astria-sequencer/src/fees/action.rs:152]: Followup issue: #1715

File: crates/astria-sequencer/src/transaction/checks.rs [hunk 2]
```diff
@@ -64,14 +48,18 @@ pub(crate) async fn get_fees_for_transaction<S: StateRead>(
     tx: &TransactionBody,
     state: &S,
 ) -> Result<HashMap<asset::IbcPrefixed, u128>> {
+    let mut fees_by_asset = HashMap::new();
```
[the remark this answers is not in the document]
Comment (ethanoroshiba) [reply crates/astria-sequencer/src/transaction/checks.rs]: Done in ee8047f.

File: crates/astria-sequencer/src/transaction/checks.rs [hunk 3]
```diff
@@ -64,14 +48,18 @@ pub(crate) async fn get_fees_for_transaction<S: StateRead>(
...
+
+    // All retrieved fees are optional: it is okay for the action to not have fees as long as it
+    // isn't part of the transaction.
```
[the remark this answers is not in the document]
Comment (ethanoroshiba) [reply crates/astria-sequencer/src/transaction/checks.rs:52]: Followup: #1717

**Pull request merged**